int32 event_id
int32 result_id
int32 extra_data
